projects/igniteui-angular/src/lib/select/select-positioning-strategy.ts

```diff
@@ -31,201 +25,231 @@ export class SelectPositioningStrategy extends ConnectedPositioningStrategy impl
+        //      Adjust the new container position with the scrolled amount.
+        // 2nd: Try position the container after the item is scrolled.
+        // 3rd: If unsuccessful, position the overlay container on TOP/ABOVE or BOTTOM/BELLOW of the input. BOTTOM/BELLOW is preferred.
+        // 4th: On page scroll, persist the ddl container position relative to its target element.
+
+        const rects = super.calculateElementRectangles(contentElement);
+        // selectFit obj, to be used for both cases of initialCall and !initialCall(page scroll/overlay repositionAll)
+        const selectFit: SelectFit = {
+            verticalOffset: this.global_yOffset,
+            horizontalOffset: this.global_xOffset,
+            targetRect: rects.targetRect,
+            contentElementRect: rects.elementRect,
+            styles: this.global_styles
         };
+
+        if (initialCall) {
+            selectFit.viewPortRect = Util.getViewportRect(document);
+
+            // Fill in the required selectFit object properties.
+            this.calculateVariables(selectFit);
+            // Calculate input and selected item elements style related variables
+            selectFit.styles = this.calculateStyles(selectFit);
+            selectFit.viewPortRect = Util.getViewportRect(document);
+
+            // Calculate how much to offset the overlay container.
+            this.calculateYoffset(selectFit);
+            this.calculateXoffset(selectFit);
+
+            // Scroll if the selected item is not fully visible and adjust the contentElement Y position with the scrolled amount
+            this.manageScrollToItem(selectFit);
+
+            super.updateViewPortFit(selectFit);
+            if (!selectFit.fitVertical) {
+                this.fitInViewport(contentElement, selectFit);
+            }
+        }
+        this.setStyles(contentElement, selectFit, initialCall);
     }
 
-    private listOutOfBounds(elementContainer: { top: number, bottom: number }, document: Document): {
-        Direction: Direction,
-        Amount: number
-    } {
-        const container = {
-            TOP: elementContainer.top,
-            BOTTOM: elementContainer.bottom,
-        };
-        const viewPort = Util.getViewportRect(document);
-        const documentElement = {
-            TOP: viewPort.top,
-            BOTTOM: viewPort.bottom
-        };
-        const returnVals = {
-            Direction: Direction.None,
-            Amount: 0
-        };
-        if (documentElement.TOP + this.defaultWindowToListOffset > container.TOP) {
-            returnVals.Direction = Direction.Top;
-            returnVals.Amount = documentElement.TOP - container.TOP;
-        } else if (documentElement.BOTTOM - this.defaultWindowToListOffset < container.BOTTOM) {
-            returnVals.Direction = Direction.Bottom;
-            returnVals.Amount = container.BOTTOM - documentElement.BOTTOM;
+    public itemIsInvisible(selectFit: SelectFit) {
```

Comment:
you should scroll if item is partially not visible, right.
To check if the item is partially not visible you can use:

```
    public itemIsInvisible(selectFit: SelectFit) {
        const itemTop = Math.round(selectFit.itemRect.top * 100) / 100;
        const itemBottom = Math.round(selectFit.itemRect.bottom * 100) / 100;
        const scrollContainerRect = this.select.scrollContainer.getBoundingClientRect();
        const scrollContainerBottom = Math.round(scrollContainerRect.bottom * 100) / 100;
        const scrollContainerTop = Math.round(scrollContainerRect.top * 100) / 100;
        // selected item is completely invisible
        return scrollContainerBottom <= itemBottom || itemTop <= scrollContainerTop;
    }

```
However I would put this code in `manageScrollToItem` method like this:

```
    private manageScrollToItem(selectFit: SelectFit) {
        const itemTop = Math.round(selectFit.itemRect.top * 100) / 100;
        const itemBottom = Math.round(selectFit.itemRect.bottom * 100) / 100;
        const scrollContainerRect = this.select.scrollContainer.getBoundingClientRect();
        const scrollContainerBottom = Math.round(scrollContainerRect.bottom * 100) / 100;
        const scrollContainerTop = Math.round(scrollContainerRect.top * 100) / 100;
        // selected item is partially or completely invisible
        const itemIsPartiallyNotVisible = scrollContainerBottom <= itemBottom || itemTop <= scrollContainerTop;

        // Scroll and compensate the item's container position, when the selected item is not visible.
        if (itemIsPartiallyNotVisible) {
            const scrollAmount = selectFit.itemElement ?
                Math.floor(selectFit.itemRect.bottom - scrollContainerRect.bottom) : 0;
            this.scrollToItem(scrollAmount);
            selectFit.verticalOffset += scrollAmount;
            this.global_yOffset = selectFit.verticalOffset;
        }
    }
```
Notice `scrollToItem` now accepts `scrollAmount` and should set `scrollTop` to whatever is passed. Also `compensateYScroll` is removed too.

Reply:
itemIsPartiallyNotVisible  --> itemIsNotCompletelyVisible. Actually it needs both checks if an item is either invisible or partially invisible.